sig/datadog/appsec/waf.rbs

```diff
@@ -3,9 +3,9 @@ module Datadog
     module WAF
       module LibDDWAF
         class Error < StandardError
-          attr_reader ruleset_info: Hash[Symbol, untyped] | nil
+          attr_reader ruleset_info: ::Datadog::AppSec::WAF::data
 
-          def initialize: (::String msg, ?ruleset_info: Hash[Symbol, untyped]?) -> void
+          def initialize: (::String msg, ?ruleset_info: ::Datadog::AppSec::WAF::data?) -> void
         end
```

Comment:
Minor: In a way, we've weakened typechecking here. Previously, we stated that this data was a Hash, and it still must be a hash (if I understand the change in libddwaf properly).

But now we say it can be anything from libddwaf, which is not actually the case. I suspect the issue may be that since we're using `object_to_ruby`, the type that falls out of it is a lot more generic than it needs to be. I wonder if there's a way to assert to the typechecker that the type for this object should be the stricter one...?

(I'm guessing this is what happens in other places where you're sprinkling `data` as well)

Reply:
I will leave that exercise for another PR 😄